### PR TITLE
change peer identity to pubkey

### LINKED ISSUE #1045: it seems it only cares about pubKey in Handshake?
https://github.com/AElfProject/AElf/blob/701e7670da7fffd84f3eabcd06f21eb4321e7e8a/AElf.OS.Core/Network/Infrastructure/IPeerPool.cs#L15-L20

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 701e7670da7fffd84f3eabcd06f21eb4321e7e8a when #1021 was merged. cc @AElfProject.

### LINKED ISSUE #1046: is it equal to address?
https://github.com/AElfProject/AElf/blob/701e7670da7fffd84f3eabcd06f21eb4321e7e8a/AElf.OS.Core/Network/Infrastructure/IPeerPool.cs#L20-L23

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 701e7670da7fffd84f3eabcd06f21eb4321e7e8a when #1021 was merged. cc @AElfProject.

### LINKED ISSUE #1121: Network test improvements
- [ ] debug skipped tests.
- [ ] cleanup event after assert.

### PR DESCRIPTION
**Pull request by samfrb**
fixes #1046, fixes #1045
fixes #1121

### CODE REVIEW

File: AElf.OS.Network.Grpc/GrpcPeerPool.cs [hunk 1]
```diff
@@ -27,6 +28,16 @@ public class GrpcPeerPool : IPeerPool
 
         public ILogger<GrpcPeerPool> Logger { get; set; }
 
+        private string _localPubKey;
+        private string LocalPublickey
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcPeerPool.cs]: should not make as a property

File: AElf.OS.Network.Grpc/GrpcPeer.cs [hunk 2]
```diff
@@ -132,7 +124,7 @@ private void HandleFailure(RpcException rpcException)
                         {
                             await StopAsync(); // shutdown for good
                         }
-                        catch (Exception e)
+                        catch (Exception)
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcPeer.cs]: do not catch exception!!!

File: AElf.OS.Network.Grpc/GrpcNetworkServer.cs [hunk 3]
```diff
@@ -63,7 +73,15 @@ public async Task StartAsync()
 
         public async Task StopAsync(bool gracefulDisconnect = true)
         {
-            await _server.KillAsync();
+            try
+            {
+                await _server.KillAsync();
+            }
+            catch (Exception)
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcNetworkServer.cs]: do not catch exception!!!

File: AElf.OS.Network.Grpc/GrpcServerService.cs [hunk 4]
```diff
@@ -21,19 +23,29 @@ namespace AElf.OS.Network.Grpc
...
         private readonly IBlockchainService _blockChainService;
+        private readonly IAccountService _accountService;
 
         public ILocalEventBus EventBus { get; set; }
 
-        public ILogger<GrpcServerService> Logger { get; set; }
+        public ILogger<GrpcServerService> Logger { get; set; }    
+        
+        private string _localPubKey;
+        private string LocalPublickey
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcServerService.cs]: no

File: AElf.OS.Network.Grpc/GrpcServerService.cs [hunk 5]
```diff
@@ -120,26 +136,25 @@ await EventBus.PublishAsync(new TransactionsReceivedEvent()
...
             }
 
             try
             {
                 Logger.LogDebug($"Received announce {an.BlockHash} from {context.Peer}.");
-                _ = EventBus.PublishAsync(new AnnouncementReceivedEventData(an,
-                    GrpcUrl.Parse(context.Peer).ToIpPortFormat()));
+                _ = EventBus.PublishAsync(new AnnouncementReceivedEventData(an, context.GetPublicKey()));
             }
             catch (Exception e)
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcServerService.cs]: no

File: AElf.OS.Network.Grpc/GrpcServerService.cs [hunk 6]
```diff
@@ -202,7 +217,7 @@ public override async Task<VoidReply> Disconnect(DisconnectReason request, Serve
         {
             try
             {
-                await _peerPool.ProcessDisconnection(GrpcUrl.Parse(context.Peer).ToIpPortFormat());
+                await _peerPool.ProcessDisconnection(context.GetPublicKey());
             }
             catch (Exception e)
```
Comment (loning) [inline AElf.OS.Network.Grpc/GrpcServerService.cs]: no

File: AElf.OS/Handlers/PeerConnectedEventHandler.cs [hunk 7]
```diff
@@ -79,15 +79,15 @@ private async Task ProcessNewBlock(AnnouncementReceivedEventData header, string
...
                     await BackgroundJobManager.EnqueueAsync(new ForkDownloadJobArgs
                     {
-                        SuggestedPeerAddress = peerAddress,
+                        SuggestedPeerPubKey = senderPubKey,
                         BlockHash = header.Announce.BlockHash.ToHex(),
                         BlockHeight = blockHeight
                     });
                 }
             }
             catch (Exception e)
```
Comment (loning) [inline AElf.OS/Handlers/PeerConnectedEventHandler.cs]: no

File: AElf.OS.Network.Grpc/AuthInterceptor.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        private readonly IPeerPool _peerPool;
+
+        public AuthInterceptor(IPeerPool peerPool)
+        {
+            _peerPool = peerPool;
+        }
+
+        public override Task<TResponse> UnaryServerHandler<TRequest, TResponse>(TRequest request, ServerCallContext context, UnaryServerMethod<TRequest, TResponse> continuation)
+        {
+            if (context.Method != "/" + nameof(PeerService) + "/" + nameof(PeerService.PeerServiceBase.Connect))
```
Comment (loning) [inline AElf.OS.Network.Grpc/AuthInterceptor.cs]: looks not good

Comment (samfrb) [reply AElf.OS.Network.Grpc/AuthInterceptor.cs]: yeah I also think it's not soo good, for now I don't have a better solution for this.

**Pull request merged**